### PR TITLE
Use the pegout creation transaction hash as the key in the map structure that stores the pegout transactions waiting for signatures

### PR DESCRIPTION
**Pull request by marcos-iov**

### CODE REVIEW

File: IPs/RSKIP375.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+|**Created**    |30-MAY-23 |
+|**Author**     |MI |
+|**Purpose**    |Sca,Usa |
+|**Layer**      |Core |
+|**Complexity** |1 |
+|**Status**     |Draft |
+
+## Abstract
+
+For the PowHSMs to sign a pegout transaction, the PowPeg node needs to provide the transaction to  sign along with the block number where the pegout was created. This information allows the PowHSM to validate that the transaction to be signed has enough PoW accumulated on top.
```
Comment (nathanieliov) [inline IPs/RSKIP375.md:25]:
```suggestion
For the PowHSMs to sign a pegout transaction, the PowPeg node needs to provide the transaction to sign along with the Rootstock block number where the pegout was created. This information allows the PowHSM to validate that the transaction to be signed has enough PoW accumulated on top.
```

**Review by nathanieliov — APPROVED** [review]
Well done. LGTM. Just a small correction.

**Review by nathanieliov — APPROVED** [review]

File: IPs/RSKIP375.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+|**Created**    |30-MAY-23 |
+|**Author**     |MI |
+|**Purpose**    |Sca,Usa |
+|**Layer**      |Core |
+|**Complexity** |1 |
+|**Status**     |Draft |
+
+## Abstract
+
+For the PowHSMs to sign a pegout transaction, the PowPeg node needs to provide the transaction to sign along with the Rootstock block number where the pegout was created. This information allows the PowHSM to validate that the transaction to be signed has enough PoW accumulated on top.
```
Comment (kelvinator07) [inline IPs/RSKIP375.md:25]:
```suggestion
For the PowHSMs to sign a pegout transaction, the PowPeg node needs to provide the transaction to sign along with the Rootstock block number where the pegout was created. This information allows the PowHSM to validate that the transaction to be signed has enough PoW(proof of work) or just Proof of work accumulated on top.
```

File: IPs/RSKIP375.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+To sign a peg-out transaction, the PowHSM needs to have registered the block where the pegout transaction was created. The PowPeg node then sends to the PowHSM the block number where the pegout was created and the transaction to sign in order to validate and sign the pegout.
+
+There is currently no easy way for the PowPeg node to obtain the Rootstock block where a pegout transaction was created. At the moment, it keeps an index in local storage that connects pegout transactions hash (without signatures) to the hash of the transaction where the pegout was created by the Bridge contract. This solution presents some disadvantages, for example, this index is kept encoded in a local file and if for any reason this file is deleted or corrupted it leaves the PowPeg incapacitated to sign pegouts.
+
+This RSKIP proposes setting the Rootstock transaction hash where a pegout is created as the key in the map structure that holds the pegouts that have already accumulated enough confirmations and are now waiting for signatures. The PowPeg node then reads this map by accessing the Bridge storage and can get from it all the information needed to supply the PowHsm to sign.
```
Comment (kelvinator07) [inline IPs/RSKIP375.md:41]:
```suggestion
This RSKIP proposes setting the Rootstock transaction hash where a pegout is created as the key in the map structure that holds the pegouts that have already accumulated enough confirmations and are now waiting for signatures. The PowPeg node then reads this map by accessing the Bridge storage and can get from it all the information needed to supply the PowHSM to sign.
```

**Pull request merged**